src/pages/Signup/Signup.jsx

```diff
@@ -1,5 +1,166 @@
 import React from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { useState } from "react";
+import styles from "./Signup.module.css";
+import axios from "axios";
 
 export default function Signup() {
-  return <div>Signup</div>;
+  const defaultData = {
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+  };
+
+  const [userData, setUserData] = useState(defaultData);
+  const [error, setError] = useState("");
+  const [showPassword, setShowPassword] = useState(false);
+  const navigate = useNavigate();
+
+  const handleInput = (e) => {
+    const { value, name } = e.target;
+    setUserData((prev) => ({ ...prev, [name]: value }));
+  };
+
+  const handleSignup = async (userData) => {
+    try {
+      const response = await axios.post("/api/auth/signup", userData);
+      if (response.status === 201) {
+        setUserData(defaultData);
+        navigate("/login");
+        console.log("HOAGAYA");
+        localStorage.setItem("Token", response.data.encodedToken);
+      }
+    } catch (error) {
+      setError("Something went wrong");
+      console.log(error);
+    }
+  };
+  const pwdVisibiltyHandler = () => {
+    setShowPassword(!showPassword);
```

Comment:
here, do a functional update instead

Reply:
Thank You , have corrected it in the latest commit